src/app/mod.rs

```diff
@@ -722,6 +722,9 @@ impl App {
         if payout.batch_id.is_some() {
             return Err(ApplicationError::PayoutAlreadyCommitted);
         }
+        if payout.is_cancelled() {
```

Comment:
I'm not sure we need this - it can simply be a no-op / early return instead of an error.